### ISSUE TITLE
Opti : Detecting specific part of content changes from modified file

### ISSUE THREAD

**Issue by tu2-atmanand** [tu2-atmanand/Task-Board#73]
## Issue

At present the only thing which is not optimized is : 
- **Real-Time Scanning is ON :** If user is making lot of changes in lot of files and there are already lot of tasks at various locations inside the file. But the user is not making any changes to the content of the task. Then, the file is being contineously scanned when the user is changing the focus from the editor. This is causing the the **Task Board** to get refresh again and again, inspite of nothing has been changed of the task content.
- **Real-Time Scanning is OFF :** In this case, its not a problem, since all the modified files are being scanned at once and also the board is getting refreshed only once.

## **Possible Solution**

After the files has been scanned, a function can be created which will basically compare the previous task and the new task detected from the current file. Only if any one of the task content has been changed, then only it will be updated in the tasksData and the board will be refreshed. But this comparing function will be costly.

Need to come up with a better solution where, we can find which part of the file has been changed and whether there is any task or task's content on that line.

**Comment by tu2-atmanand** [comment]
I created a similar issue for this exact problem : 

https://github.com/tu2-atmanand/Task-Board/issues/83

**Comment by tu2-atmanand** [comment]
> You might find something using following : https://docs.obsidian.md/Reference/TypeScript+API/ListItemCache 

 _Originally posted by @tu2-atmanand in [#83](https://github.com/tu2-atmanand/Task-Board/issues/83#issuecomment-2556375159)_
